hack/license/gen/main.go

```
@@ -178,101 +180,117 @@ func readAndRewrite(path string) error {
-				_, err = buf.WriteString(line)
-				if err != nil {
-					log.Fatal(err)
-				}
-				_, err = buf.WriteString("\n")
-				if err != nil {
-					log.Fatal(err)
-				}
-				continue
-			}
-			if lf && strings.HasPrefix(line, d.Escape) {
-				continue
-			} else if !bf {
-				once.Do(func() {
-					err = apache.Execute(buf, d)
-					if err != nil {
-						log.Fatal(err)
-					}
-				})
-				lf = false
-			}
-			if !lf {
-				_, err = buf.WriteString(line)
-				if err != nil {
-					log.Fatal(err)
-				}
-				_, err = buf.WriteString("\n")
-				if err != nil {
-					log.Fatal(err)
-				}
-			}
-			bf = false
-		}
-	}
-	err = f.Close()
-	if err != nil {
-		return errors.Errorf("filepath %s, could not close", path)
+
+		process(sc, ext, buf, d)
 	}
+
 	err = os.RemoveAll(path)
 	if err != nil {
 		return errors.Errorf("filepath %s, could not delete", path)
 	}
+
 	f, err = os.Create(path)
 	if err != nil {
-		err = f.Close()
-		if err != nil {
+		if err := f.Close(); err != nil {
 			log.Fatal(err)
 		}
 		return errors.Errorf("filepath %s, could not open", path)
 	}
-	_, err = f.WriteString(strings.ReplaceAll(buf.String(), d.Escape+"\n\n\n", d.Escape+"\n\n"))
-	if err != nil {
+	defer func() {
+		if err = f.Close(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	if _, err = f.WriteString(strings.ReplaceAll(buf.String(), d.Escape+"\n\n\n", d.Escape+"\n\n")); err != nil {
 		log.Fatal(err)
 	}
-	err = f.Close()
-	if err != nil {
-		log.Fatal(err)
+	if err := f.Sync(); err != nil {
+		return err
 	}
+
 	return nil
 }
 
+func process(in *bufio.Scanner, ext string, out *bytes.Buffer, d Data) {
+	haveHeader := false
+	var line string
+
+	// write header string to output
+	for in.Scan() {
+		line = in.Text()
+		if !isBuildFlag(ext, line) {
+			if strings.HasPrefix(line, d.Escape) {
+				continue
+			}
+			break
+		}
+
+		if _, err := out.WriteString(line); err != nil {
+			log.Fatal(err)
```

`log.Fatal` is called a lot of places. Should we return an error? 🤔